### ISSUE TITLE
训练时，logits.float()新申请的显存数过大导致oom

### ISSUE THREAD

**Issue by yyyhhl** [modelscope/ms-swift#6966]
训练参数：local bs=1, max_length=32k, zero3_offload
模型：qwen3-30b-a3b-instruct-2507，
训练方法：继续预训练：
卡：8机*8卡A100

请问logits.float()这个49G哪来？logits.float()这一步应该是把序列的logits从fp16转为fp32，最新需要分配的总显存应该是32k*151936(vocab_size)*4/(1024^3)=18.5G，怎么会有49G呢？

<img width="3456" height="1334" alt="Image" src="https://github.com/user-attachments/assets/9ed1c204-c5e8-42ca-8d54-15df7460e344" />

<img width="3456" height="1392" alt="Image" src="https://github.com/user-attachments/assets/7f158fa7-a98a-46d0-845d-b32d6224ab3d" />

<img width="3456" height="2006" alt="Image" src="https://github.com/user-attachments/assets/2c2224b4-39b5-46c1-bc29-8ed2f221f0b7" />

**Comment by Jintao-Huang** [comment]
尝试使用megatron-swift训练

https://swift.readthedocs.io/zh-cn/latest/Megatron-SWIFT/Mcore-Bridge.html#moe

**Comment by yyyhhl** [comment]
> 尝试使用megatron-swift训练
> 
> https://swift.readthedocs.io/zh-cn/latest/Megatron-SWIFT/Mcore-Bridge.html#moe

那这里是49g不是18.5g的原因方便说一下嘛？

**Comment by yyyhhl** [comment]
> 尝试使用megatron-swift训练
> 
> https://swift.readthedocs.io/zh-cn/latest/Megatron-SWIFT/Mcore-Bridge.html#moe
另外请教下为什么继续预训练时，use_logits_to_keep会是True呢？如果是true的话只保留最后一个token的logits，更不应该是49g了？
<img width="3456" height="1720" alt="Image" src="https://github.com/user-attachments/assets/30ad46e0-2867-4664-8c72-56eb7878f00d" />
